README.md

```diff
@@ -629,7 +629,30 @@ choices = [
 ]
 ```
 
-You can specify `:key` as an additional option which will be used as short name for selecting the choice via keyboard key press.
+You can specify `:key` as an additional option which will be used as short name for selecting the choice via keyboard key press. When used with `:enum`, the key is displayed instead of a number.
+
+```ruby
+choices = [{name: "small", key: "s"}, {name: "medium", key: "m"}, {name: "large", key: "l"}]
+prompt.select("What size?", choices, enum: ')')
+# =>
+# What size? (Press ↑/↓ arrow to move and Enter to select)
+# ‣ s) small
+#   m) medium
+#   l) large
+```
+
+When providing `:key`, you can override the displayed text with `:key_name` - as in the case of using the escape key:
+
+
+```ruby
+choices = [{name: "next", key: "n"}, {name: "previous", key: "p"}, {name: "exit", key: :escape, key_name: "esc"}]
+prompt.select("Do what?", choices, enum: ')')
+# =>
+# Do what? (Press ↑/↓ arrow to move and Enter to select)
+# ‣ n) next
+#   p) previous
+#   esc) exit
```

Comment:
I feel as though the `:enum` should still act as an enumerated list of items, even when combined with key selection. Potentially the display could resolve this by appending key afterwards like this:

```
# =>
# Do what? (Press ↑/↓ arrow to move and Enter to select)
# ‣ 1) next [n]
#   2) previous [p]
#   3) exit [esc]
```

Reply:
I agree, this could be side-by-side, and would likely be a lot more intuitive (i.e. you expect `:enum` to add the numbers, but for some reason `:key` overrides that)

Reply:
I agree that this can/should be split up into more manageable chunks. As far as display, the options I see are a) as you said, an appended `[key]` text, or b) underlining, like alt-key hotkey representation in many OSes. The second would be tricky in a situation where you want the `:key` to be a letter that is not in the word, or up/downcase version of it (i.e. <ins>f</ins>ish, <ins>F</ins>ish, fi<ins>S</ins>h).

As for non-`select` usage, I had not considered that to be honest - it should definitely be supported, and I agree that `:key_action => select` really only makes sense for select. Maybe there can be a specific `Menu` class that works more like an interactive menu system, which is what I intended this for, versus adding that functionality into just `select`. 

So, you'd like the changes split like this:

1. A PR for the `Choice`/`Choices` changes and related tests.
2. The ability to pres to move to an item in select` , and "check" item(s) in `multi_select`
3. The "interactive menu" stuff, i.e. `:key_action => select`

If that all sounds good to you, I can get 1 & 2 done and we can talk more about 3.

Reply:
The a) suggestion seems to me a bit more straightforward in a sense that some keys don't directly correspond with letters like `esc` key. I like the idea of underlined letters but I worry about how the support looks in various terminals. Often the underline is not displayed at all. In general the b) option feels a bit more fragile to me.

There may be an opportunity to create some common menu abstraction. I definitely want to pursue an idea of splitting up the whole library into plugins. It would remain a single gem but all the menus would rely on common interfaces and hence provide a way for people to create their own prompts or enhance the current ones.

The plan sounds good to me. Could I be cheeky and suggest an item 0? When you made changes to slider prompt, you suggested that the `default` apart from index could accept a name. I like this idea and would extend it to `select`, `multi_select` and `enum_select` prompts as well. This could be then released with the slider changes. What do you think?

Reply:
> In general the b) option feels a bit more fragile to me

Agreed. While it would be cooler looking, it definitely comes with a lot more complication. a) it is.

> There may be an opportunity to create some common menu abstraction. I definitely want to pursue an idea of splitting up the whole library into plugins.

This would be neat. Making the "prompt" things more modular would be beneficial. If you need help with that, you know how to find me :)

> extend it to select, multi_select and enum_select prompts as well

Ah, of course - I forgot about that! I will definitely do that as well!

Reply:
Hi Katelyn, it's been a while since we had our last discussion about this feature. In the meantime, I've updated the `default` option for all the prompts and made some changes to the `Choice`. I wonder if you have time to revisit adding key support again?